packages/babel-parser/src/parser/expression.js

```diff
@@ -935,12 +917,7 @@ export default class ExpressionParser extends LValParser {
       }
 
       elts.push(
-        this.parseExprListItem(
-          false,
-          refExpressionErrors,
-          { start: 0 },
```

Comment:
Q: `refNeedsArrowPos` is reset here. Do we need to reset `refExpressionErrors.optionalParameters` to `-1`? Maybe worth a test covering this.